src/containers/mode-tools.jsx

```diff
@@ -13,10 +13,126 @@ class ModeTools extends React.Component {
     constructor (props) {
         super(props);
         bindAll(this, [
+            '_getSelectedUncurvedPoints',
+            '_getSelectedUnpointedPoints',
+            'hasSelectedUncurvedPoints',
+            'hasSelectedUnpointedPoints',
             'handleCopyToClipboard',
-            'handlePasteFromClipboard'
+            'handleCurvePoints',
+            'handlePasteFromClipboard',
+            'handlePointPoints'
         ]);
     }
+    _getSelectedUncurvedPoints () {
+        const items = [];
+        const selectedItems = getSelectedLeafItems();
+        for (const item of selectedItems) {
+            if (!item.segments) continue;
+            for (const seg of item.segments) {
+                if (seg.selected) {
+                    const prev = seg.getPrevious();
+                    const next = seg.getNext();
+                    const isCurved =
+                        (!prev || seg.handleIn.length > 0) &&
+                        (!next || seg.handleOut.length > 0) &&
+                        (prev && next ? seg.handleOut.isColinear(seg.handleIn) : true);
+                    if (!isCurved) items.push(seg);
+                }
+            }
+        }
+        return items;
+    }
+    _getSelectedUnpointedPoints () {
+        const points = [];
+        const selectedItems = getSelectedLeafItems();
+        for (const item of selectedItems) {
+            if (!item.segments) continue;
+            for (const seg of item.segments) {
+                if (seg.selected) {
+                    if (seg.handleIn.length > 0 || seg.handleOut.length > 0) {
+                        points.push(seg);
+                    }
+                }
+            }
+        }
+        return points;
+    }
+    hasSelectedUncurvedPoints () {
+        const points = this._getSelectedUncurvedPoints();
+        return points.length > 0;
+    }
+    hasSelectedUnpointedPoints () {
+        const points = this._getSelectedUnpointedPoints();
+        return points.length > 0;
+    }
+    handleCurvePoints () {
+        let changed;
+        const points = this._getSelectedUncurvedPoints();
+        for (const point of points) {
+            const prev = point.getPrevious();
+            const next = point.getNext();
+            const noHandles = point.handleIn.length === 0 && point.handleOut.length === 0;
+            if (!prev && !next) {
+                continue;
+            } else if (prev && next && noHandles) {
+                // Handles are parallel to the line from prev to next
+                point.handleIn = prev.point.subtract(next.point)
+                    .normalize()
+                    .multiply(prev.getCurve().length / 2);
```

Comment:
Since the length of the resulting handles is largely arbitrary, I think we should use a value that has a neat geometric property: if you have a rectangle/circle, converting all the points to curved turns it into an oval/circle. 
Right now half the path length does this
![handles-too-large](https://user-images.githubusercontent.com/654102/34301290-fc846b26-e6f9-11e7-9e12-d2f011342b6c.gif)

But we could make it do this
![handles-just-right](https://user-images.githubusercontent.com/654102/34301453-9f4dc6c2-e6fa-11e7-90de-332e0ed3bf77.gif)

Right now we are scaling the prev curve length by 0.5, but there is a magic number to get close to a circle: `0.3902628565`. 

_/warning math_
Arrive at that by looking at how to approximate a circle with bezier curves [source](http://spencermortensen.com/articles/bezier-circle/). A unit circle is approximated by:
```
P_0 = (0,1), P_1 = (c,1), P_2 = (1,c), P_3 = (1,0)
c = 0.551915024494
```
We are trying to turn a square (side length `s`) into a circle with the same anchors, which will be a larger circle with radius `s * sqrt(2) / 2`. So the correct scaling for the new handles is `c * sqrt(2) / 2`, or `0.3902628565`. Turns out this works with rectangles => ovals also, although I haven't done the math on why...

I checked with Carl, and FWIW this is actually the behavior from other vector editors like Sketch as well.

Reply:
Sounds good! .5 always felt a bit too long.